techmatters/hrm
# Implement Edit Contact Permissions

Note: https://github.com/techmatters/hrm/pull/187 is merged into this PR. Hold until that one is merged to review. 
Primary reviewer: @stephenhand 

## Description
This PR adds the backend support for edit contacts permissions, and starts using it in the edit contact endpoint.
Contact actions so far can make use of 3 conditions: `isSupervisor` and `everyone` same as the other permissions, and a new one, `isOwner` is introduced. Given an user and a contact, `isOwner` will evaluate to true if the user is the `twilioWorkerId` value of the contact.

@dee-luo in this PR I'm giving `[['isSupervisor'], ['isOwner']]` rules to all helplines on `editContact`. Lemme know if there are different requirements for this.

### Checklist
- [x] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-1198)

### Verification steps
- Start development server and expose your localhost via http tunnel (with ngrok like `ngrok http 3000 `).
- Edit the current permissions config accordingly (add `isOwner` if intended to test).
- Start Flex dev server, enable edit contact feature flag and point hrm url to the above tunnel.
- Try editing a contact of your own (succeed) and one from somebody else while not being supervisor (fail).
- Repeat above while being supervisor (both should fail).
- If desired try different combinations of the permissions rules for the `editContact` action.
